app/lib/actions/pulp/repository/refresh.rb

```diff
@@ -9,6 +9,7 @@ class Refresh < Pulp::Abstract
         def plan(repository, input = {})
           update_or_associate_importer(input[:capsule_id], repository)
           update_or_associate_distributors(input[:capsule_id], repository)
+          remove_unnecessary_distributors(input[:capsule_id], repository)
```

Comment:
One small optimization, maybe pass the output of 'retrieve_with_details()' into each function, it looks like its called in all 3